libraries/cmd/unit_generic.h

```diff
@@ -165,6 +165,7 @@ class Unit : public Armed, public Audible, public Drawable, public Damageable, p
     // - model (stock) - should be variant, but only if actually different without upgrades
     // TODO: name is duplicated down below in some memory saving measure (StringPool::Reference)
     std::string unit_key;
+    bool is_planet{};
```

Comment:
Better to make it

`enum Vega_UnitType unit_type`

and then set it at every constructor and be done. That would also mean you don't need `isUnit` anymore.

Reply:
That's a good point. `unit_type` makes more sense. Although `isUnit` might be part of the public, Python-facing interface of the Unit class, so I don't think I will remove or rename that method now. But the field underneath can be `unit_type`.

Reply:
OK. So in the base `Unit` class, I have a field named `unit_type`. It's private. It is only used in the base class. Every subclass hard-codes their unit type. In the new, virtual method named `getUnitType()`. (In the base `Unit` class, of course, `getUnitType()` returns the value of `unit_type`.)

Note that `getUnitType()` and `isUnit()` are functionally identical. They always return the same thing. It's just a matter of which one makes the most logical sense to use in a particular context. In almost all cases, getUnitType seems to me to make more sense, so I have converted most of the 100 or so references to `isUnit` to `getUnitType` instead.